packages/modules/providers/payment-stripe/src/core/stripe-base.ts

```diff
@@ -95,221 +113,271 @@ abstract class StripeBase extends AbstractPaymentProvider<StripeOptions> {
 
-    if (stripeId !== data.customer) {
-      return await this.initiatePayment(input)
-    } else {
-      if (isPresent(amount) && data.amount === amountNumeric) {
-        return { data }
-      }
+      return { data: sessionData }
+    } catch (e) {
+      throw this.buildError("An error occurred in updatePayment", e)
+    }
+  }
+
+  async createAccountHolder({
+    context,
+  }: CreateAccountHolderInput): Promise<CreateAccountHolderOutput> {
+    const { accountHolder, customer } = context
+
+    if (accountHolder?.data?.id) {
+      return { id: accountHolder.data.id as string }
+    }
 
-      try {
-        const id = data.id as string
-        const sessionData = (await this.stripe_.paymentIntents.update(id, {
-          amount: amountNumeric,
-        })) as unknown as PaymentProviderSessionResponse["data"]
+    if (!customer) {
+      throw this.buildError(
+        "No customer in context",
+        new Error("No customer provided while creating account holder")
+      )
+    }
+
+    const shipping = customer.billing_address
+      ? ({
+          address: {
+            city: customer.billing_address.city,
+            country: customer.billing_address.country_code,
+            line1: customer.billing_address.address_1,
+            line2: customer.billing_address.address_2,
+            postal_code: customer.billing_address.postal_code,
+            state: customer.billing_address.province,
+          },
+        } as Stripe.CustomerCreateParams.Shipping)
+      : undefined
 
-        return { data: sessionData }
-      } catch (e) {
-        return this.buildError("An error occurred in updatePayment", e)
+    try {
+      const stripeCustomer = await this.stripe_.customers.create({
+        email: customer.email,
+        name:
+          customer.company_name ||
+          `${customer.first_name ?? ""} ${customer.last_name ?? ""}`.trim() ||
+          undefined,
+        phone: customer.phone as string | undefined,
+        ...shipping,
+      })
+
+      return {
+        id: stripeCustomer.id,
+        data: stripeCustomer as unknown as Record<string, unknown>,
       }
+    } catch (e) {
+      throw this.buildError(
+        "An error occurred in createAccountHolder when creating a Stripe customer",
+        e
+      )
     }
   }
 
-  async listPaymentMethods(
-    context: PaymentProviderContext
-  ): Promise<PaymentMethodResponse[]> {
-    const customerId = context.customer?.metadata?.stripe_id
-    if (!customerId) {
+  async deleteAccountHolder({
+    context,
+  }: DeleteAccountHolderInput): Promise<DeleteAccountHolderOutput> {
+    const { accountHolder } = context
+    const accountHolderId = accountHolder?.data?.id as string | undefined
+    if (!accountHolderId) {
+      throw this.buildError(
+        "No account holder in context",
+        new Error("No account holder provided while deleting account holder")
+      )
+    }
+
+    try {
+      await this.stripe_.customers.del(accountHolderId)
```

Comment:
question: Wouldn't this need to be the `external_id` of the account holder?

Reply:
Yes, we just get it from `data.id` since `data` is whatever the payment provider returns. We could also get it from `externalId` but that is not required in the context, that is just stored in our table for convenience

Reply:
Oh yeah, I see. Makes sense